File: Algorithms/C++/705-Design_HashSet.cpp
// Runtime: 96 ms, faster than 83.28% of C++ online submissions for Design HashSet.
// Memory Usage: 76.6 MB, less than 33.33% of C++ online submissions for Design HashSet.
class MyHashSet {
public:
    /** Initialize your data structure here. */
    MyHashSet() {
        memset(arr, 0, 1000001);
    }
    
    void add(int key) {
        arr[key] = true;
    }
    
    void remove(int key) {
        arr[key] = false;
    }
    
    /** Returns true if this set contains the specified element */
    bool contains(int key) {
        return arr[key];
    }
private:
    int arr[1000001];
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */

File: Algorithms/Python3.x/705-Design_HashSet.py
# Runtime: 284 ms, faster than 38.56% of Python3 online submissions for Design HashSet.
# Memory Usage: 39.7 MB, less than 7.69% of Python3 online submissions for Design HashSet.
class MyHashSet:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.array = [False] * (10 ** 6 + 1)

    def add(self, key: int) -> None:
        self.array[key] = True

    def remove(self, key: int) -> None:
        self.array[key] = False

    def contains(self, key: int) -> bool:
        """
        Returns true if this set contains the specified element
        """
        return self.array[key]


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)
